content/blog/2024/10/what-is-the-esp-registry/index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: "What is the ESP-Registry?"
+date: 2024-10-27
+showAuthor: false
+authors:
+  - "pedro-minatel"
+tags: ["Registry", "Component", "ESP-IDF"]
+---
+
+When starting a new project, we often reuse code from existing components, such as libraries and drivers, to save time. This not only saves time but also makes your project easier to maintain.
+
+In this article, we will explore what the ESP-Registry is and the key concepts of components, making it easy to understand not only how to use components but also how to contribute.
+
+## Concept
+
+In the [ESP-IDF](https://idf.espressif.com), components are widely used to make the framework more modular, making it clearer to understand and maintain. The naming convention in the ESP-IDF is to use the term "component" instead of "library" because the ESP-IDF is more like an operating system with self-contained modules that interact with the system at a higher level.
+
+There are three concepts that we need to introduce in order to understand the complete solution:
+
+### 1. Component
+
+As mentioned before, the component is a self-contained piece of code that can be used to add a new functionality to the application. In most of the cases, the component is used in several applications, making easy to reuse code and as well will improve the efficiency and reduce the development complexity.
+
+For example, in your application (`my-application`), you are going to use a specific sensor (`sensor-component`) that will collect temperature and humidity from the environment. Instead of adding all the logic behind the communication protocol, data collection, conversions and etc to the application code, you can add as a component.

```

Comment:
```suggestion
For example, let's imagine there is an application called `my-application` in which we want to use a specific sensor that will collect temperature and humidity from the environment. Instead of adding all the logic behind the communication protocol, data collection, conversions, etc. to the application code, you can add a component `sensor-component`, that has this logic already implemented in code. The important part is that this code has been reviewed and tested multiple times by other developers. 
```